### PR TITLE
Fix `new collective` name input elements

### PR DESCRIPTION
**Pull request by susnux**
### 📝 Summary

* Resolves: #548 

This resolves some styling issues, like hovering the submit input does not set the border on the text input or mismatching borders. Moreover the `NcMultiselect` is deprecated and should be replaced with the `NcSelect`.

#### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
only submit button has error color, wrong border radius | both elements input and submit have correct borders
hovering only highlights the submit button | correct border color on whole element on hover
. | submit button with correct borders
circle select separated from submit button | submit button part of circle select
multselect no elements | NcSelect no results

Please note that the missing primary color border on hover will be fixed with the next vue components update (see https://github.com/nextcloud/nextcloud-vue/pull/3701 )

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [ ] Tests (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [x] Documentation ([README](https://github.com/nextcloud/collectives/blob/main/README.md) or [documentation](https://github.com/nextcloud/collectives/blob/main/docs/)) has been updated or is not required

### CODE REVIEW

**Comment by max-nextcloud** [comment]
I'd suggest to still merge this. It fixes ui bugs. If the ui is replaced even before the next release so be it. Maybe the modal takes longer than expected, maybe we want to use this input again later on. Even if not - i don't see any downside to merging this. But maybe I am missing something, @mejo-

File: src/components/Nav/NewCollective.vue [hunk 1]
```diff
@@ -33,51 +33,60 @@
 			</NcActionButton>
 		</template>
 	</NcAppNavigationItem>
-	<div v-else class="collective-create">
-		<form v-show="editing" @submit.prevent.stop="createCollective">
```
Comment (mejo-) [inline src/components/Nav/NewCollective.vue:37]: Removing the `<form>` element with its submit event means that the form cannot be submitted by pressing <kbd>Return</kbd> anymore.

File: src/components/Nav/NewCollective.vue [hunk 2]
```diff
@@ -33,51 +33,60 @@
...
+				:title="t('collectives', 'Select emoji')"
+				class="button-emoji"
+				@click.prevent>
+				{{ emoji }}
+			</NcButton>
+		</NcEmojiPicker>
+
+		<NcTextField v-if="!pickCircle"
+			:value.sync="text"
+			:error="!text"
```
Comment (mejo-) [inline src/components/Nav/NewCollective.vue:49]: Not sure whether I like that the input field is marked as erroneous from the very start. Visual feedback is nice, but it feels wrong to me to have the error border displayed before even starting to type something.

File: src/components/Nav/NewCollective.vue [hunk 3]
```diff
@@ -33,51 +33,60 @@
...
+			:label="t('collectives', 'New collective name')"
+			:show-trailing-button="!!text"
+			required
+			trailing-button-icon="arrowRight"
+			:trailing-button-label="t('collectives', 'Create new collective')"
+			@trailing-button-click="createCollective" />
+
+		<div v-else class="circle-selector--container">
+			<NcSelect ref="circleSelector"
```
Comment (mejo-) [inline src/components/Nav/NewCollective.vue:58]: Not sure whether it's the switch from `NcMultiSelect` to `NcSelect`, but when selecting a circle with a longer name, the select field size grows larger than it should. Also the cancel and expand/collapse buttons inside the select field are not properly aligned.

**Review by mejo- — CHANGES_REQUESTED** [review]
After some first testing, I identified the following issues.

File: src/components/Nav/NewCollective.vue (hunk 1 above)
Comment (susnux) [reply src/components/Nav/NewCollective.vue:37]: Reverted, does work now :)

File: src/components/Nav/NewCollective.vue (hunk 2 above)
Comment (susnux) [reply src/components/Nav/NewCollective.vue:49]: Me neither... decided to restore the previous behavior and only show the error state if the user started typing

File: src/components/Nav/NewCollective.vue (hunk 3 above)
Comment (susnux) [reply src/components/Nav/NewCollective.vue:62]: I can not confirm this, for me (Firefox 109, collectives on current NC master) it works as expected and looks like this:
image

Comment (mejo-) [reply src/components/Nav/NewCollective.vue:62]: Ah, and it only grows bigger when closing the dropdown. With the dropdown open it keeps its desired size, like in your screenshot.

Comment (mejo-) [reply src/components/Nav/NewCollective.vue:62]: Also, for some reason the selection is dark even when I use the light theme :thinking:

Comment (mejo-) [reply src/components/Nav/NewCollective.vue:62]: > Also, for some reason the selection is dark even when I use the light theme thinking

Nevermind this. It seems to be a bug in nextcloud-vue. At least all other NcSelect fields are dark as well for me when my browser prefers dark theme but Nextcloud is set to light theme.

Comment (susnux) [reply src/components/Nav/NewCollective.vue:62]: Yes this is known and already fixed, waiting for release: https://github.com/nextcloud/nextcloud-vue/pull/3707

Comment (susnux) [reply src/components/Nav/NewCollective.vue:62]: > Ah, and it only grows bigger when closing the dropdown. With the dropdown open it keeps its desired size, like in your screenshot.

I pushed a commit which fixes parts of this issue. The other  parts are fixed with: https://github.com/nextcloud/nextcloud-vue/pull/3759

> Nevermind this. It seems to be a bug in nextcloud-vue. At least all other NcSelect fields are dark as well for me when my browser prefers dark theme but Nextcloud is set to light theme.

Fixed if we upgrade `@nextcloud/vue` to the new released `7.6.0`.

**Review by mejo- — APPROVED** [review]

**Comment by mejo-** [comment]
Thanks a lot @susnux :heart:

**Pull request merged**